LOGGING/src/NLogTarget/ApplicationInsightsTarget.cs

```diff
@@ -118,10 +129,24 @@ protected override void InitializeTarget()
             this.telemetryClient = new TelemetryClient();
 #pragma warning restore CS0618 // Type or member is obsolete
 
-            string instrumentationKey = this.instrumentationKeyLayout.Render(LogEventInfo.CreateNullEvent());
-            if (!string.IsNullOrWhiteSpace(instrumentationKey))
+            string connectionString = this.connectionStringLayout.Render(LogEventInfo.CreateNullEvent());
+           
+            // Check if nlog application insights target has connectionstring in config file then
+            // configure telemetryclient with the connectionstring otherwise using instrumentationkey.
+            if (!string.IsNullOrWhiteSpace(connectionString))
             {
-                this.telemetryClient.Context.InstrumentationKey = instrumentationKey;
+                var configuration = TelemetryConfiguration.CreateDefault();
+                configuration.ConnectionString = connectionString;
+
+                this.telemetryClient.TelemetryConfiguration.ConnectionString = connectionString;
```

Comment:
This could be tricky... Will this affect the TC.Active? Or this will only affect the telemetryclient owned by nlogtarget? The ikey approach only affected own telemetryclient, and nothing else.

Reply:
@cijothomas - this.telemetryClient field is local for this Microsoft.ApplicationInsights.NLogTarget.ApplicationInsightsTarget class. Is there any other code to refer to for the iKey approach?